src/game_initialization/lobby_data.cpp

```diff
@@ -462,7 +462,7 @@ game_info::game_info(const config& game, const std::vector<std::string>& install
 
 game_info::ADDON_REQ game_info::check_addon_version_compatibility(const config& local_item, const config& game)
 {
-	if(!local_item.has_attribute("addon_id") || !local_item.has_attribute("addon_version")) {
+	if(!local_item.has_attribute("addon_id") || !local_item.has_attribute("addon_version") || !local_item["required"].to_bool(false)) {
```

Comment:
this looks wrong, i don't see why local_item should have an attribute `required`

Reply:
It looks like I put it in the wrong place (and typoed the attribute name) - I've moved it to https://github.com/wesnoth/wesnoth/pull/4884/commits/b87c5f8472c416fc45ed3d6a869d909ccbcb7406#diff-a1267dbf4e2d22a4ec5b65914f0606adR218